spyder_kernels/customize/spyderpdb.py

```diff
@@ -705,6 +705,26 @@ def notify_spyder(self):
 
         get_ipython().kernel.publish_pdb_state(step)
 
+        # Publish Pdb state so we can update the Variable Explorer
+        # and the Editor on the Spyder side
+        pdb_stack = traceback.StackSummary.extract(self.stack)
+        pdb_index = self.curindex
+
+        skip_hidden = getattr(self, 'skip_hidden', False)
+
+        if skip_hidden:
+            # Filter out the hidden frames
+            hidden = self.hidden_frames(self.stack)
+            pdb_stack = [f for f, h in zip(pdb_stack, hidden) if not h]
+            # Adjust the index
+            pdb_index -= sum(hidden[:pdb_index])
+
+        try:
+            frontend_request(blocking=False).set_pdb_stack(
+                pdb_stack, pdb_index)
```

Comment:
Do you need to do adjustments here after PR #381? Just asking because I'm not really sure.

Reply:
No I think PR #381 improves this one, but because we are adding to the end of the function everything is good